nemo_skills/pipeline/utils.py

```diff
@@ -699,8 +696,9 @@ def add_task(
 
     Note that there are two parameters that control dependencies.
         - task_dependencies: list of tasks that this task depends on **within the same experiment**
-        - run_after: a single **experiment name** that this task should run after. Will schedule
+        - run_after: an **experiment name** that this task should run after. Will schedule
           dependencies on all tasks inside `run_after` experiment. It needs to already be launched and running.
+          Can also use multiple names separated with ||
```

Comment:
I guess that's better, let me make a change. The reason why I prefer not to use list is that it makes cmdlines more cumbersome, since Typer only supports lists if you explicitly use --arg-name multiple times (so --run_after exp1 --run_after exp2 ... ) which is imo worse than what you could do with argparse using (--run_after exp1 exp2 ...). Actually now that I'm writing this I think we can just split on space same as we do for some other list args that we need to handle as space is most likely not a valid value for slurm. Do you think it's better that way or would prefer a list as an explicit type?

Reply:
My concern was primarily that, anywhere I want to use this with the python API, I basically have to go through and merge my experiment names with the separator. Pythonically, it would feel more ergonomic to have this type be a list.

You raise a good point about it being something you would want to be able to handle from the command line as well.

I think in principle, it generally makes sense to handle command-line parsing logic as close to the command line as possible (with this util being in a file that is never directly called from command-line, it is "far" from the command line code).

It looks like Typer supports custom [typer parsers](https://typer.tiangolo.com/tutorial/parameter-types/custom-types/#type-parser). This would allow you to tell the command line args how to parse the string (e.g. into a list) by adding a `parser` argument to the `typer.Option` call.

IMO the latter is the most correct thing to do--but it would also require us to import and use the parser everywhere as well, slightly increasing the complexity.

Reply:
cool, I agree. In any case I think our usage becomes more and more through python so putting python convenience first makes sense. And it makes even more sense given that we can use the custom types in the future to make cli behave as we want